examples/hetero/hetero_link_pred.py

```diff
@@ -0,0 +1,219 @@
+import os.path as osp
+import argparse
+import torch
+from torch.nn import Parameter
+import torch.nn.functional as F
+
+from torch_geometric.datasets import MovieLens
+from torch_geometric.transforms import ToUndirected, RandomLinkSplit
+
+from torch_geometric.nn import SAGEConv, to_hetero
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '--feature_extraction_model', type=str, default='all-MiniLM-L6-v2',
+    help='Model used to transform movie titles to node features.')
+parser.add_argument(
+    '--link_split_num_val', type=float, default=0.1)
+parser.add_argument(
+    '--link_split_num_test', type=float, default=0.1)
+parser.add_argument(
+    '--disjoint_train_ratio', type=float, default=0.0,
+    help=('Proportion of training edges used for supervision. '
+          'The remainder is disjoint and used for message passing.'
+          'This option can be used to avoid target leakage.')
+)
+parser.add_argument('--num_epochs', type=int, default=101)
+parser.add_argument('--lr', type=float, default=0.01)
+parser.add_argument('--weight_decay', type=float, default=0.0005)
+parser.add_argument('--hidden', type=int, default=32)
+args = parser.parse_args()
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '../../data/MovieLens')
+dataset = MovieLens(path, model_name=args.feature_extraction_model)
+data = dataset[0]
+
+data['user'].x = torch.eye(data['user'].num_nodes, dtype=data['movie'].x.dtype)
+
+# We can now convert `data` into an appropriate format for training a
+# graph-based machine learning model:
+
+# 1. Add a reverse ('movie', 'rev_rates', 'user') relation for message passing.
+data = ToUndirected()(data)
+del data['movie', 'rev_rates', 'user'].edge_label  # Remove "reverse" label.Fd
```

Comment:
I'll remove the stray "Fd" at the end here but I don't want to keep changing the code until you've had a chance to take a look as well.